cartridges/int_algolia/cartridge/scripts/algolia/helper/logHelper.js

```diff
@@ -158,8 +160,61 @@ function setLogData(id, productLog) {
     return true;
 }
 
+/**
+ * Logs the object IDs of problematic objects.
+ *
+ * If the input is an array, it logs the object IDs of all objects in the array that have a `body.objectID` property.
+ * If the input is an object, it logs the object's `objectID` property if it exists.
+ *
+ * @param {Object|Object[]} object - The object or array of objects to log the object IDs of.
+ * @param {Error} error - The error that occurred while processing the object.
+ */
+function customLogger(object, error) {
+    if (!object) {
+        logger.error('Error: ' + error);
+        return;
+    }
+
+    var requests = object.requests;
+    var problematicObjectIDs = [];
+
+    if (Array.isArray(requests)) {
+        requests.forEach(function (item) {
+            if (item && item.body && item.body.objectID) {
+                problematicObjectIDs.push(item.body.objectID);
+            }
+            logAttributes(item);
+        });
+    } else if (requests && requests.objectID) {
+        problematicObjectIDs.push(requests.objectID);
+        logAttributes(requests);
+    } else {
+        logger.error('Error: ' + error);
+        return;
+    }
+
+    logger.error('Error: ' + error + 
+        ' Invalid request body. Problematic objectIDs: ' + problematicObjectIDs.join(', '));
+}
+
+/**
+ * Logs the attributes of an object.
+ * @param {Object} item - The object to log the attributes of.
+ */
+function logAttributes(item) {
+
+    var objAttr = [];
+    for (var key in item) {
+        if (item.hasOwnProperty(key)) {
+            objAttr.push(key + ': ' + item[key]);
+        }
+    }
+    logger.error('Invalid request body. Problematic Object attributes: ' + objAttr.join(', '));
```

Comment:
we are keeping the problematic body in line 193 and this line's purpose is keeping which action causing the issue

Reply:
I understand the intention, but you can see that it will not produce something useful in reality. As for this PR, it would be nice to have a simple try/catch that reports the problem without making the job failing yes, but not something too complex and noisy like that.
Feel free to discard for now.